sdk/aqueduct/api_client.py

```diff
@@ -98,9 +102,16 @@ def __init__(self, api_key: str, aqueduct_address: str):
         else:
             self.use_https = self._test_connection_protocol(try_http=True, try_https=True)
 
-    def _construct_full_url(self, route_suffix: str, use_https: bool) -> str:
+    def _construct_base_url(self, use_https: Optional[bool] = None) -> str:
+        if use_https == None:
+            use_https = self.use_https
```

Comment:
lets try to use `is None` comparisons here: https://stackoverflow.com/questions/14247373/python-none-comparison-should-i-use-is-or#:~:text=is%20is%20generally%20preferred%20when,and%20even%20on%20their%20ordering.

Reply:
also, shouldn't we error here instead of using https? `use_https` must have been set by the constructor.

Reply:
We can get rid of the `use_https` argument right?

Reply:
`use_https` is set to `self.use_https` except for in `_test_connection_protocol` which sets the argument directly to True or False

Reply:
ah I remember now, ok sg.

Reply:
^ Because of that, I don't really see a need to explicitly add `use_https=self.use_https` in the function call each time, since the function has access to that value so I changed it to optional rather than required